cpp/src/Ice/ConnectionI.cpp

```diff
@@ -2512,6 +2534,36 @@ Ice::ConnectionI::initiateShutdown()
     }
 }
 
+bool
+Ice::ConnectionI::idleCheck() noexcept
+{
+    std::lock_guard lock(_mutex);
+    if (_state == StateActive)
+    {
+        if (_transceiver->hasDataAvailable())
+        {
+            return true;
+        }
+        else
+        {
+            // TODO: replace by ConnectionIdleException.
+            setState(StateClosed, make_exception_ptr(TimeoutException(__FILE__, __LINE__)));
+        }
+    }
+    return false;
```

Comment:
How does this work? The connection doesn't become idle while there is data available, because we want to give a chance to the thread pool for reading this data.

What is the meaning of the return value?

Reply:
Reworked and commented.